src/Core/Content/Media/Infrastructure/Path/MediaUrlGenerator.php

```diff
@@ -38,4 +40,43 @@ public function generate(array $paths): array
 
         return $urls;
     }
+
+    private function encodeUrl(string $mediaUrl): string
+    {
+        if (!Feature::isActive('v6.8.0.0')) {
+            return $mediaUrl;
+        }
+
+        $urlInfo = parse_url($mediaUrl);
+        if (!\is_array($urlInfo)) {
+            return $mediaUrl;
+        }
+
+        $segments = explode('/', $urlInfo['path'] ?? '');
+
+        foreach ($segments as $index => $segment) {
+            $segments[$index] = rawurlencode($segment);
+        }
+
+        $path = implode('/', $segments);
+        if (isset($urlInfo['query'])) {
+            $path .= "?{$urlInfo['query']}";
+        }
+
+        $encodedPath = '';
+
+        if (isset($urlInfo['scheme'])) {
+            $encodedPath = "{$urlInfo['scheme']}://";
+        }
+
+        if (isset($urlInfo['host'])) {
+            $encodedPath .= "{$urlInfo['host']}";
+
+            if (isset($urlInfo['port'])) {
+                $encodedPath .= ":{$urlInfo['port']}";
+            }
+        }
+
+        return $encodedPath . $path;
+    }
```

Comment:
Is this the same code as here?

https://github.com/shopware/shopware/blob/32c45e2b237cb5661a41e629984330ce047ac3a8/src/Storefront/Framework/Twig/Extension/UrlEncodingTwigFilter.php#L25-L62

Then I would suggest making one of them easily accessible (e.g. making this `public static`) and then calling this function in the twig filter.